src/main/CommandHandler.cpp

```diff
@@ -626,47 +626,54 @@ CommandHandler::connect(std::string const& params, std::string& retStr)
 void
 CommandHandler::dropPeer(std::string const& params, std::string& retStr)
 {
-    std::map<std::string, std::string> retMap;
-    http::server::server::parseParams(params, retMap);
-
-    auto peerId = retMap.find("node");
-    auto ban = retMap.find("ban");
-    if (peerId != retMap.end())
+    try
     {
-        auto found = false;
-        NodeID n;
-        if (mApp.getHerder().resolveNodeID(peerId->second, n))
+        std::map<std::string, std::string> retMap;
+        http::server::server::parseParams(params, retMap);
+
+        auto peerId = retMap.find("node");
+        auto ban = retMap.find("ban");
+        if (peerId != retMap.end())
         {
-            auto peers = mApp.getOverlayManager().getPeers();
-            auto peer = std::find_if(
-                peers.begin(), peers.end(),
-                [&n](Peer::pointer peer) { return peer->getPeerID() == n; });
-            if (peer != peers.end())
+            auto found = false;
+            NodeID n;
+            if (mApp.getHerder().resolveNodeID(peerId->second, n))
             {
-                mApp.getOverlayManager().dropPeer(*peer);
-                if (ban != retMap.end() && ban->second == "1")
+                auto peers = mApp.getOverlayManager().getPeers();
+                auto peer = std::find_if(
+                    peers.begin(), peers.end(),
+                    [&n](Peer::pointer peer) { return peer->getPeerID() == n; });
+                if (peer != peers.end())
                 {
-                    retStr = "Drop and ban peer: ";
-                    mApp.getBanManager().banNode(n);
+                    mApp.getOverlayManager().dropPeer(*peer);
+                    if (ban != retMap.end() && ban->second == "1")
+                    {
+                        retStr = "Drop and ban peer: ";
+                        mApp.getBanManager().banNode(n);
+                    }
+                    else
+                        retStr = "Drop peer: ";
+
+                    retStr += peerId->second;
+                    found = true;
                 }
-                else
-                    retStr = "Drop peer: ";
+            }
 
+            if (!found)
+            {
+                retStr = "Peer ";
                 retStr += peerId->second;
-                found = true;
+                retStr += " not found";
             }
         }
-
-        if (!found)
+        else
         {
-            retStr = "Peer ";
-            retStr += peerId->second;
-            retStr += " not found";
+            retStr = "Must specify at least peer id: droppeer?node=NODE_ID";
         }
     }
-    else
+    catch (std::exception& e)
```

Comment:
Just a try catch in Config::resolveNodeID right? That returns false if it throws.